assets/js/modules/analytics/components/common/UseUAandGA4SnippetSwitches.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Analytics Use UA and GA4 Snippet Switch component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from '../../datastore/constants';
+import { MODULES_ANALYTICS_4 } from '../../../analytics-4/datastore/constants';
+import Switch from '../../../../components/Switch';
+import { trackEvent } from '../../../../util';
+const { useSelect, useDispatch } = Data;
+
+export default function UseUAandGA4SnippetSwitches() {
+	const useUASnippet = useSelect( ( select ) => select( STORE_NAME ).getUseSnippet() );
+	const canUseUASnippet = useSelect( ( select ) => select( STORE_NAME ).getCanUseSnippet() );
+	const useGA4Snippet = useSelect( ( select ) => select( MODULES_ANALYTICS_4 ).getUseSnippet() );
+
+	const { setUseSnippet: setUseUASnippet } = useDispatch( STORE_NAME );
+	const { setUseSnippet: setUseGA4Snippet } = useDispatch( MODULES_ANALYTICS_4 );
+
+	const onUAChange = useCallback( () => {
+		setUseUASnippet( ! useUASnippet );
+		trackEvent( 'analytics_setup', useUASnippet ? 'analytics_tag_enabled' : 'analytics_tag_disabled' );
+	}, [ useUASnippet, setUseUASnippet ] );
+
+	const onGA4Change = useCallback( () => {
+		setUseGA4Snippet( ! useGA4Snippet );
+	}, [ useGA4Snippet, setUseGA4Snippet ] );
+
+	if ( useGA4Snippet === undefined || useUASnippet === undefined ) {
+		return null;
+	}
+
+	let message;
+
+	if ( useUASnippet && useGA4Snippet ) {
+		message = 'Site Kit will add the Universal Analytics and Google Analytics 4 codes automatically';
+	} else if ( useUASnippet && ! useGA4Snippet ) {
+		message = 'Site Kit will add the Universal Analytics code automatically';
+	} else if ( ! useUASnippet && useGA4Snippet ) {
+		message = 'Site Kit will add the Google Analytics 4 code automatically';
+	} else {
+		message = 'Site Kit will not add the code to your site';
+	}
```

Comment:
These strings all need to be localisable:

```suggestion
	if ( useUASnippet && useGA4Snippet ) {
		message = __( 'Site Kit will add the Universal Analytics and Google Analytics 4 codes automatically', 'google-site-kit' );
	} else if ( useUASnippet && ! useGA4Snippet ) {
		message = __( 'Site Kit will add the Universal Analytics code automatically', 'google-site-kit' );
	} else if ( ! useUASnippet && useGA4Snippet ) {
		message = __( 'Site Kit will add the Google Analytics 4 code automatically', 'google-site-kit' );
	} else {
		message = __( 'Site Kit will not add the code to your site', 'google-site-kit' );
	}
```